common/reflect_resource_plugin_framework.go

```diff
@@ -0,0 +1,505 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Converts a tfsdk struct into a go-sdk struct.
+func TfSdkToGoSdkStruct(tfsdk interface{}, gosdk interface{}, ctx context.Context) error {
```

Comment:
Suggestion: can we return `diag.Diagnostics` from this? Then in resources/datasources calling this method, we can use the same
```go
diags.append(TfSdkToGoSdkStruct(...))
if diags.HasError() {
  return
}
```
pattern that is encouraged by the plugin framework.